backend/inspirehep/records/api/base.py

```diff
@@ -451,3 +454,49 @@ def serialize_for_es(self):
     def get_value(self, field, default=None):
         """Method which makes ``get_value`` more intuitive"""
         return get_value(self, field, default)
+
+    def revert(self, revision_id):
+        if self.model is None:
+            raise MissingModelError()
+
+        revision = self.revisions[revision_id]
+
+        with db.session.begin_nested():
+            before_record_revert.send(current_app._get_current_object(), record=self)
+
+            self.update(dict(revision))
+
+        after_record_revert.send(current_app._get_current_object(), record=self)
+        return self
+
+    @property
+    def revisions(self):
+        """Get Inspire revisions iterator
+
+        Replaces RevisionsIterator from Invenio until they merge proper patch
+        """
+        if self.model is None:
+            raise MissingModelError()
+
+        return InspireRevisionsIterator(self.model)
+
+
+class InspireRevisionsIterator(RevisionsIterator):
+    def __getitem__(self, revision_id):
+        """Get a specific revision.
+
+        Revision id is always smaller by 1 from version_id.
+        Replaces RevisionsIterator from Invenio until they merge proper patch
+        """
+        if revision_id < 0:
```

Comment:
shouldn't this be `< 1`?

Reply:
no, as usually revision 0 is correct. But in Inspire it's impossible to have revision 0 as we are doing increment by 2 when we create record.